pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -222,6 +239,37 @@ func (f *framework) RunPrefilterPlugins(
 	return nil
 }
 
+// RunScorePlugins runs the set of configured scoring plugins. It returns a map that
+// stores for each scoring plugin name the corresponding NodeScoreList(s).
+// It also returns *Status, which is set to non-success if any of the plugins returns
+// a non-success status.
+func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.Node) (PluginToNodeScoreMap, *Status) {
+	pluginToNodeScoreMap := make(PluginToNodeScoreMap, len(f.scorePlugins))
+	for i := range f.scorePlugins {
```

Comment:
Similar to how we run our priority functions today, we should start one go routine to run all score plugins for a node. In your current implementation we start one go routine per score plugin. Since most of these score plugins finish quickly, starting a go routine per score plugin may impose a significant overhead.